Lib/pdb.py

```diff
@@ -414,16 +416,27 @@ def preloop(self):
                     self.message('display %s: %r  [old: %r]' %
                                  (expr, newvalue, oldvalue))
 
-    def interaction(self, frame, traceback):
+    def interaction(self, frame, tb_or_exc):
         # Restore the previous signal handler at the Pdb prompt.
+
+        if isinstance(tb_or_exc, BaseException):
+            tb_or_exc, exception = tb_or_exc.__traceback__, tb_or_exc
+            self._chained_exceptions = [exception]
+            current = exception
+            while current := current.__context__:
+                self._chained_exception_index += 1
+                self._chained_exceptions.insert(0, current)
```

Comment:
What about ``current.__cause__``?

Reply:
My understanding is that `__context__` is always set, and `__cause__ == __context__` if `raise Err(..) from e`, so I _think_ we only need to look at `__context__`. 

```python
In [5]: try:
    ...:   try:
    ...:     raise ValueError('A')
    ...:   except Exception as e:
    ...:     raise TypeError('B') from e
    ...: except Exception as f:
    ...:     print(f.__context__, f.__cause__)
A A

In [6]: try:
    ...:   try:
    ...:     raise ValueError('A')
    ...:   except Exception as e:
    ...:     raise TypeError('B')
    ...: except Exception as f:
    ...:     print(f.__context__, f.__cause__)
A None
```

I originally was looking at both, but removed as `__context__` appeared to be enough. 
Unless we do want to have a visual distinction between when there is `__cause__` or not. 

Unless there is a case where we can have `__cause__` but not `__context__`, or where both are different, but I could not think of one. Though i'm far from an expert so if you think it's necessary I can check both.

Reply:
The cause is whatever was in the "raise .. from". See below.

```
% cat xx.py 

import traceback

try:
    try:
        raise TypeError("bad type")
    except:
        raise ValueError("bad value") from OSError("something happened")
except Exception as e:
    exc = e

print("------------------- exc:")
traceback.print_exception(exc)
print("------------------- context:")
traceback.print_exception(exc.__context__)
print("------------------- cause:")
traceback.print_exception(exc.__cause__)
```


```
% ./python.exe xx.py
------------------- exc:
OSError: something happened

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/iritkatriel/src/cpython-1/xx.py", line 8, in <module>
    raise ValueError("bad value") from OSError("something happened")
ValueError: bad value
------------------- context:
Traceback (most recent call last):
  File "/Users/iritkatriel/src/cpython-1/xx.py", line 6, in <module>
    raise TypeError("bad type")
TypeError: bad type
------------------- cause:
OSError: something happened

```

Reply:
Ah ! Thanks ! This is helpful. 

I've reworked the code to work with both `__cause__` and `__context__`.